### PR TITLE
feat: Include images in call to LLM

### PR DESCRIPTION
**Pull request by cecheta**
Required for #750

## Purpose

* Pass retrieved images to LLM when advanced image processing is enabled

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How to Test
* Run tests
* Deploy with advanced image processing enabled, upload images, ask questions based on images

### CODE REVIEW

File: code/backend/batch/utilities/tools/question_answer_tool.py [hunk 1]
```diff
@@ -122,20 +125,41 @@ def generate_on_your_data_messages(
             },
             *QuestionAnswerTool.clean_chat_history(chat_history),
             {
-                "content": self.config.prompts.answering_user_prompt.format(
-                    sources=documents,
-                    question=question,
-                ),
+                "content": [
```
Comment (cecheta) [inline code/backend/batch/utilities/tools/question_answer_tool.py:128]: This format works even when not using vision

**Comment by cecheta** [comment]
Note that we ran into this issue where the LLM attempts to provide an answer from every image passed to the LLM, even if it's not relevant

Screenshot 2024-05-21 145846

This should be looked into during #964

**Comment by adamdougal** [comment]
> Note that we ran into this issue where the LLM attempts to provide an answer from every image passed to the LLM, even if it's not relevant
> 
> Screenshot 2024-05-21 145846
> 
> This should be looked into during #964

Is this something to do with `top_k`? Or is this something we can tweak in the system prompt to say ignore anything not relevant?

**Review by adamdougal — DISMISSED** [review]
lgtm

File: code/backend/batch/utilities/tools/question_answer_tool.py [hunk 2]
```diff
@@ -145,8 +169,39 @@ def answer_question(self, question: str, chat_history: list[dict], **kwargs):
...
+        )
+
+        return clean_answer
+
+    def create_image_url_list(self, source_documents):
+        image_types = self.config.get_advanced_image_processing_image_types()
+
+        image_documents = []
+
+        for doc in source_documents:
```
Comment (ross-p-smith) [inline code/backend/batch/utilities/tools/question_answer_tool.py:184]: Apologies, I'm reading this in the browser. But it seems we iterate around one list and then iterate around it again to add the sas token - could we do this all in one iteration?

Comment (cecheta) [reply code/backend/batch/utilities/tools/question_answer_tool.py]: Good point, done

**Comment by cecheta** [comment]
> > Note that we ran into this issue where the LLM attempts to provide an answer from every image passed to the LLM, even if it's not relevant
> > Screenshot 2024-05-21 145846
> > This should be looked into during #964
> 
> Is this something to do with `top_k`? Or is this something we can tweak in the system prompt to say ignore anything not relevant?

I think it would need to be done in the prompt

**Review by adamdougal — APPROVED** [review]

**Pull request merged**